### PR TITLE
#995 add android viewtag for espresso driver

### PR DESCRIPTION
**Pull request by saikrishna321**
## Change list

Related : appium/appium-espresso-driver#189
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/io/appium/java_client/MobileBy.java [hunk 1]
```diff
@@ -561,6 +571,69 @@ protected ByImage(String b64Template) {
...
+            if (FindsByAndroidViewTag.class.isAssignableFrom(contextClass)) {
+                return FindsByAndroidViewTag.class.cast(context)
+                        .findElementByAndroidViewTag(getLocatorString());
+            }
+
+            if (FindsByFluentSelector.class.isAssignableFrom(contextClass)) {
+                return super.findElement(context);
+            }
+
+            throw formIllegalArgumentException(contextClass, FindsByAccessibilityId.class,
```
Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/MobileBy.java]: `FindsByAndroidViewTag`

File: src/main/java/io/appium/java_client/MobileBy.java [hunk 2]
```diff
@@ -121,6 +121,16 @@ public static By windowsAutomation(final String windowsAutomation) {
         return new ByWindowsAutomation(windowsAutomation);
     }
 
+    /**
+     * This locator strategy is available in Espresso Driver mode.
+     * @since Appium 1.8.2 beta
+     * @param tag is an an view tag string
```
Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/MobileBy.java]: `an`

File: src/main/java/io/appium/java_client/android/AndroidDriver.java [hunk 3]
```diff
@@ -24,11 +24,7 @@
 
 import com.google.common.collect.ImmutableMap;
 
-import io.appium.java_client.AppiumDriver;
-import io.appium.java_client.CommandExecutionHelper;
-import io.appium.java_client.FindsByAndroidUIAutomator;
-import io.appium.java_client.HasOnScreenKeyboard;
-import io.appium.java_client.LocksDevice;
+import io.appium.java_client.*;
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/AndroidDriver.java]: As far as I remember the linter does like star imports

Comment (saikrishna321) [reply src/main/java/io/appium/java_client/android/AndroidDriver.java]: Fixed!

**Pull request merged**